charts/templates/configmap-script.yaml

```diff
@@ -0,0 +1,663 @@
+
+                if [[ "$certType" == "self-signed" ]]; then
+                log "Info: [poststart] Generate temporary certificate if necessary"
+                $AUTH_CURL -k -X POST -i -d @createTempCert.xqy -H "Content-type: application/x-www-form-urlencoded" \
+                -H "Accept: multipart/mixed; boundary=BOUNDARY" https://localhost:8000/v1/eval
+                resp_code=$?
+                info "response code for Generate temporary certificate is $resp_code"
+            fi
+        fi
+        
+        log "Info: [poststart] removing cert keys"
+        rm -f /run/secrets/marklogic-certs/*.key
+    }   
+
+    ###############################################################
+    # Env Setup of MarkLogic
+    ###############################################################
+    # Make sure username and password variables are not empty
+    if [[ -z "${MARKLOGIC_ADMIN_USERNAME}" ]] || [[ -z "${MARKLOGIC_ADMIN_PASSWORD}" ]]; then
+        error "MARKLOGIC_ADMIN_USERNAME and MARKLOGIC_ADMIN_PASSWORD must be set." exit
+    fi
+
+    # generate JSON payload conditionally with license details.
+    if [[ -z "${LICENSE_KEY}" ]] || [[ -z "${LICENSEE}" ]]; then
+        LICENSE_PAYLOAD="{}"
+    else
+        info "LICENSE_KEY and LICENSEE are defined, installing MarkLogic license."
+        LICENSE_PAYLOAD="{\"license-key\" : \"${LICENSE_KEY}\",\"licensee\" : \"${LICENSEE}\"}"
+    fi
+
+    # sets realm conditionally based on user input
+    if [[ -z "${REALM}" ]]; then
+        ML_REALM="public"
+    else
+        info "REALM is defined, setting realm."
+        ML_REALM="${REALM}"
+    fi
+
+    if [[ -z "${MARKLOGIC_WALLET_PASSWORD}" ]]; then
+        MARKLOGIC_WALLET_PASSWORD_PAYLOAD=""
+    else
+        MARKLOGIC_WALLET_PASSWORD_PAYLOAD="wallet-password=${MARKLOGIC_WALLET_PASSWORD}"
+    fi
+
+    ###############################################################
+    info "Starting to Configuring MarkLogic"
+    info "HOST_FQDN: $HOST_FQDN"
+    info "MARKLOGIC_BOOTSTRAP_HOST: $MARKLOGIC_BOOTSTRAP_HOST"
+
+    # Wait for current pod ready
+    wait_until_marklogic_ready $HOST_FQDN
+
+    ###############################################################
+    # Installing admin user and init Security DB and objects
+    ###############################################################  
+
+    # Only do this if the bootstrap host is in the statefulset we are configuring
+    info "target bootstrap host: ${MARKLOGIC_BOOTSTRAP_HOST}"
+    if [[ "${MARKLOGIC_CLUSTER_TYPE}" = "bootstrap" && "${HOST_FQDN}" = "${MARKLOGIC_BOOTSTRAP_HOST}" ]]; then
+        sleep 2s
+        init_security_db
+        configure_group
+    else
+        # TODO: Finish here
+        configure_group
+        join_cluster $HOST_FQDN
+    fi
+
+    info "after init_security_db"
+    sleep 5s 
+    if [[ $MARKLOGIC_JOIN_TLS_ENABLED == "true" ]]; then
+        configure_tls
+    fi
+
+    info "helm script completed"
+
```

Comment:
When do we manage the joining process ?

Reply:
The joining process is in join_cluster function